### PR TITLE
[memory][scs-helper] Add system identifiers to output structure

### PR DESCRIPTION
**Pull request by NikitaZotov**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/CONTRIBUTING.md)
* [x] Update changelog
* [x] Update documentation

Fixed https://github.com/ostis-ai/sc-machine/issues/207

### CODE REVIEW

File: sc-memory/sc-memory/sc_scs_helper.cpp [hunk 1]
```diff
@@ -53,22 +57,23 @@ class StructGenerator
       auto const & edge = parser.GetParsedElement(t.m_edge);
       auto const & trg = parser.GetParsedElement(t.m_target);
 
-      ScAddr const srcAddr = ResolveElement(src);
-      ScAddr const trgAddr = ResolveElement(trg);
+      auto const srcAddrResult = ResolveElement(src);
+      auto const trgAddrResult = ResolveElement(trg);
```
Comment (SHtress) [inline sc-memory/sc-memory/sc_scs_helper.cpp:61]: Why auto?

File: sc-memory/sc-memory/sc_scs_helper.cpp [hunk 2]
```diff
@@ -53,22 +57,23 @@ class StructGenerator
...
 
-      SC_ASSERT(srcAddr.IsValid() && trgAddr.IsValid(), ());
```
Comment (SHtress) [inline sc-memory/sc-memory/sc_scs_helper.cpp:59]: Why remove addr validation?

File: sc-memory/sc-memory/sc_scs_helper.cpp [hunk 3]
```diff
@@ -95,9 +100,27 @@ class StructGenerator
     }
   }
 
-  void SetSCsGlobalIdtf(std::string const & idtf, ScAddr const & addr)
+  ScAddrVector SetSystemIdtf(std::string const & idtf, ScAddr const & addr)
```
Comment (KovalM) [inline sc-memory/sc-memory/sc_scs_helper.cpp:103]: After this change, we will have two ways of system idtf generation: this and m_ctx->HelperSetSystemIdtf. As result, we should support this two ways.

I think, we should leave only one way

**Review by KovalM — CHANGES_REQUESTED** [review]

File: sc-memory/sc-memory/sc_scs_helper.cpp [hunk 4]
```diff
@@ -28,8 +28,12 @@ class StructGenerator
     , m_fileInterface(std::move(fileInterface))
     , m_outputStructure(outputStructure)
   {
+    m_kNrelSysIdtf = m_ctx.HelperResolveSystemIdtf("nrel_system_identifier", ScType::NodeConstNoRole);
+    SC_ASSERT(m_kNrelSysIdtf.IsValid(), ("Keynode `nrel_system_identifier` is not valid"));
```
Comment (smousy) [inline sc-memory/sc-memory/sc_scs_helper.cpp:32]: Why you add assert instead of exception?

**Review by smousy — CHANGES_REQUESTED** [review]

**Review by KovalM — APPROVED** [review]

**Review by SHtress — APPROVED** [review]

**Review by SHtress — APPROVED** [review]

File: sc-memory/sc-memory/sc_memory.cpp [hunk 5]
```diff
@@ -446,20 +447,65 @@ ScAddr ScMemoryContext::HelperResolveSystemIdtf(std::string const & sysIdtf, ScT
...
-      bool isSuccess = HelperSetSystemIdtf(sysIdtf, resultAddr);
-      SC_ASSERT(isSuccess, ());
+      EraseElement(resultAddr);
+      resultAddr = ScAddr::Empty;
     }
   }
   return resultAddr;
 }
 
+bool ScMemoryContext::HelperResolveSystemIdtf(
```
Comment (KovalM) [inline sc-memory/sc-memory/sc_memory.cpp:459]: Let's leave single implementation of HelperResolveSystemIdtf

**Review by KovalM — CHANGES_REQUESTED** [review]

**Review by smousy — APPROVED** [review]

**Review by KovalM — APPROVED** [review]

**Pull request merged**